proposals/hips/hip-1139.md

```diff
@@ -0,0 +1,75 @@
+# Hiero SDK Design Doc: Immutable Topic Ids and Updatable Submit Keys without an Admin Key
+
+## Summary
+
+See HIP-1139: [Enable Immutable Topic Ids and Updatable Submit Keys without an Admin Key](https://github.com/hiero-ledger/hiero-improvement-proposals/blob/main/HIP/hip-1139.md)
+
+## Language
+- "Dead Key" refers to an all-zeros Ed25519 public key that is cryptographically impossible to sign with.
+- "Topic Immutability" refers to a state where both the Admin Key and Submit Key are unusable, ensuring no future modifications or operations are possible, but the Topic remains in Consensus Nodes (and block nodes in the future).
+
+## Test Plan
+
+1. **Given** a private topic is created with both an Admin Key and Submit Key, **when** the Submit Key is updated to a dead key by the Admin Key, **then** no further messages can be submitted to the topic.
+
+2. **Given** a private topic is created with both an Admin Key and Submit Key, **when** the Admin Key is updated to a dead key by itself, **then** then continue submit messages with the submit key and no further administrative updates can be made to the topic.
+
+3. **Given** a private topic is created with both keys, **when** both Submit Key and Admin Key are updated to dead keys, **then** the topic becomes fully immutable (neither messages nor updates are possible).
+
+4. **Given** a private topic is created with only a Submit Key and no Admin Key, **when** a TopicUpdateTransaction is attempted to update the Submit Key, **then** the update is successful and no more messages can be submitted.
+
+5. **Given** a public topic is created with an Admin Key but no Submit Key, **when** the Admin Key is updated to a dead key, **then** the topic becomes administratively immutable but still allows message submission.
+
+6. **Given** a public topic with a Submit Key set to a dead key, **when** a message submission is attempted using the original Submit Key, **then** the submission fails with an invalid signature or unauthorized error.
+
+7. **Given** a TopicUpdateTransaction attempts to update an Submit Key to a dead key without a valid Submit Key signature, **then** the update fails with an invalid signature.
+
+8. **Given** a TopicUpdateTransaction attempts to update an Admin Key to a dead key without a valid Admin Key signature, **then** the update fails with an invalid signature.
+
+9. **Given** a TopicUpdateTransaction attempts to update the Submit Key to a dead key with a valid Admin Key signature, **then** the update succeeds and is irreversible.
```

Comment:
Does this need a valid submit key signature as well? Or does it just need one of the admin key or submit key to sign?